app/ports/templates/ports/stats.html

```diff
@@ -139,97 +112,62 @@
-                                <td>{{ port.num }}</td>
-                            </tr>
-                        {% endfor %}
-                    </table>
-                </div>
-                <div class="col-lg-6 col-md-6 p-2">
-                    <h6 class="text-center">Requested Installations</h6>
-                    <table class="table table-condensed border">
-                        <thead>
-                        <tr>
-                            <th scope="col">Port</th>
-                            <th scope="col">Installations</th>
-                        </tr>
-                        </thead>
-                        {% for port in req_port_distribution %}
-                            <tr>
-                                <td><a href="{% url 'port_detail' port.port %}">{{ port.port }}</a></td>
-                                <td>{{ port.num }}</td>
-                            </tr>
-                        {% endfor %}
-                    </table>
-                </div>
-            </div>
+        <div class="col-lg-5">
+            <p class="text-center" style="font-size: 13px"><i>Opt-in for submitting stats by installing the port <span class="bg-light">mpstats-gsoc</span>.</i></p>
+            <br>
+            <table class="table table-striped">
+                <tbody>
+                <tr>
+                    <th scope="row">Total submissions:</th>
+                    <td class="text-right">{{ total_submissions }}</td>
+                </tr>
+                <tr>
+                    <th scope="row">Total unique users:</th>
+                    <td class="text-right">{{ unique_users }}</td>
+                </tr>
+                <tr>
+                    <th scope="row">Unique users last week:</th>
+                    <td class="text-right">{{ last_week }}</td>
+                </tr>
+                <tr>
+                    <th scope="row">Unique users this week:</th>
+                    <td class="text-right">{{ current_week }}</td>
+                </tr>
+                </tbody>
+            </table>
         </div>
     </div>
+    <br>
+    <hr>
+    <br>
+    {% csrf_token %}
+    <form class="form-inline justify-content-center" method="get" action="{% url 'stats_home' %}#os-chart">
+        <label for="days"><h5>The charts below show stats for users who made submissions in last: </h5></label>
+        <select onchange="this.form.submit()" class="form-control ml-2" name="days" id="days">
+            <option class="format-option" value="7" >7 Days</option>
+            <option class="format-option" value="30" {% if days == 30 %}selected{% endif %}>30 Days</option>
+            <option class="format-option" value="90" {% if days == 90 %}selected{% endif %}>90 Days</option>
+            <option class="format-option" value="180" {% if days == 180 %}selected{% endif %}>180 Days</option>
+            <option class="format-option" value="365" {% if days == 365 %}selected{% endif %}>365 Days</option>
```

Comment:
Isn't this code already generalized elsewhere?

Reply:
Yes, thanks.